pkg/kubelet/network/kubenet/bridge.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubenet
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"regexp"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/util"
+)
+
+var cidrRegexp = regexp.MustCompile(`inet ([0-9a-fA-F.:]*/[0-9]*)`)
+
+func createBridge(name string) error {
```

Comment:
> If we eventually thunk down to CNI's bridge driver, why do we have all this logic for bridge manipulation?

The goal at this stage was not to completely rewrite the 'cbr0' logic from Kube core, but only to move it to a plugin so that we could deprecate & remove the internal logic.  Then begin to rewrite it.  One step at a time makes easier review and testing :)